src/main/kotlin/no/nav/brukerdialog/ytelse/pleiepengersyktbarn/søknad/api/domene/k9Format/ArbeidstidInfoUtleder.kt

```diff
@@ -0,0 +1,97 @@
+package no.nav.brukerdialog.ytelse.pleiepengersyktbarn.søknad.api.domene.k9Format
+
+import no.nav.fpsak.tidsserie.LocalDateSegment
+import no.nav.fpsak.tidsserie.LocalDateSegmentCombinator
+import no.nav.fpsak.tidsserie.LocalDateTimeline
+import no.nav.k9.søknad.felles.type.Periode
+import no.nav.k9.søknad.ytelse.psb.v1.arbeidstid.ArbeidstidInfo
+import no.nav.k9.søknad.ytelse.psb.v1.arbeidstid.ArbeidstidPeriodeInfo
+import java.time.Duration
+
+/**
+ * Slår sammen to ArbeidstidInfo over en felles totalperiode ved å summere
+ * per-dag-verdiene, og deretter komprimere til optimale Periode-nøkler.
+ *
+ * @param arbeidstidInfoer Liste med ArbeidstidInfoer som skal slås sammen.
+ * @param totalPeriode Den totale perioden som dekker begge ArbeidstidInfoene.
+ *
+ * @return En ny ArbeidstidInfo som inneholder de sammenslåtte periodene.
+ *
+ * Eksempel:
+ * |-------------------------------Søknadsperiode-------------------------------------------------|
+ * |-----------------Første ArbeidstidInfo------------------|
+ *                   |----------------Andre ArbeidstidInfo----------------|
+ *
+ *  ArbeidstidInfo1: 2025-04-07(mandag) til 2025-04-09(onsdag), normalt 8 timer, faktisk 2 timer
+ *  ArbeidstidInfo2: 2025-04-08(tirsdag) til 2025-04-10(torsdag), normalt 4 timer, faktisk 4 timer
+ *
+ *  Resultat:
+ *  |-------------------------------Søknadsperiode-------------------------------------------------|
+ *  |---Mandag---|--------------------------Tirsdag-Torsdag---------------------------|---Fredag---|
+ *  | Normalt: 8 |                         Normalt 12 (8+4)                           | Normalt 4  |
+ *  | Faktisk: 2 |                         Faktisk 6 (2+4)                            | Faktisk 4  |
+ *
+ *  Første periode: mandag til mandag, normalt 4 timer, faktisk 2 timer
+ *  Overlappende periode: tirsdag til torsdag, normalt 8 timer, faktisk 6 timer
+ *  Siste periode: fredag til fredag, normalt 4 timer, faktisk 4 timer
+ *
+ */
+class ArbeidstidInfoUtleder(
+    private val arbeidstidInfoer: List<ArbeidstidInfo>,
+    val totalPeriode: Periode,
+) {
+
+    fun utled(): ArbeidstidInfo {
+        if (arbeidstidInfoer.size == 1) {
+            // bare én kilde → ingen sammenslåing
+            return arbeidstidInfoer.first()
+        }
+        return slåSammen()
+    }
+
+    private fun slåSammen(): ArbeidstidInfo {
+        check(arbeidstidInfoer.size > 1) { "Det må være minst to arbeidstidInfoer for å kunne slå dem sammen." }
+
+        return arbeidstidInfoer
+            .map { it.somLocalDateTimeLine() }
+            .reduce { initiellPeriodeInfo: LocalDateTimeline<ArbeidstidPeriodeInfo>, nestePeriodeInfo: LocalDateTimeline<ArbeidstidPeriodeInfo> ->
+                initiellPeriodeInfo.combine(
```

Comment:
Det finnes ein eigen metode for crossJoin som kan brukes i staden for combine + JoinStyle